Readme.md

```diff
@@ -12,61 +14,66 @@
-- Focus on integration tests, and testing complex pieces of code.
-- Although there is no comprehensive list, tech debt tickets are tracked in Trello with a label, or the title prefix 'Tech debt:'. Most are in the 'Someday' column. Its worth browsing them for context.
-- The fishbelt form page is a gong show. We havent had a refactor cycle to make it more maintaible in between the many various features and bandaids being added to it. Dont copy its approach. The main issue with it is that Formik didnt age will with the requirements as they revealed themselves. The observations table part of the form, thankfully, doesnt use Formik, however using two different approaches for form state cascaded into much more duplication and complication in the code for that page. For some reason the form also loses its dirty/unsaved values when the network status changes, which resulted in a hack that involves saving unsaved values in local storage, which in hindsight was a bad choice (I probably thought it was unavoidable due to props changing, but there are indications that it may be caused by internal state changing). Yuck. Here be dragons.
+- Styles use Styled Components for easy scoping, speed, and maintainability. Make sure to import using the macro for easier debugging (it results in more human-friendly classnames) `import styled from 'styled-components/macro'`
+- Focus on user-focused integration tests, and testing complex pieces of code. 100% test coverage is not a goal for this project.
+- Although there is no comprehensive list, tech debt tickets are tracked in Trello with a label, or the title prefix 'Tech debt:'. Most are in the 'Someday' column.
+- Collect Record form pages use two different approaches to managing form state. Formik had its limitations, so for the observations tables we opted to store form state with a reducer. This inconsistency has lead to some tech debt and complexity that was determined to be acceptable, but its worth knowing when handling things like dirty form state, that there are two states to consider.
 
-### General Architecture (Incomplete WIP)
+### General Architecture
 
-The general approach to this code has been to avoid premature performance optimization in favour of more maintainable code. When possible, it is also attempted to keep as much code outside of React and in regular JavaScript as possible. The idea is to make the code more approachable for non-React devs (Dustin, Kim), and to make more of the code reusable in future cases of framework swapouts.
+The general approach to this code has been to avoid premature performance optimization in favour of more maintainable code. When possible, attempts are made to keep as much code outside of React and in regular JavaScript as possible. The idea is to make the code more approachable for non-React devs, and to make more of the code reusable in future cases of framework swapouts or upgrades.
```

Comment:
are we expecting framework swapouts? has this been done in the past? I would argue to stick with react-based code if we are using react (which I believe we are doing with some exceptions)